plugins/device/deviceutils/store-utils.go

```diff
@@ -42,8 +41,20 @@ func GetDevices(peerIds ...string) ([]deviceapi.Info, error) {
 	return devices, nil
 }
 
-// DeviceExist checks the given device existence
-func DeviceExist(reqDevice string, devices []deviceapi.Info) bool {
+// FetchDevices returns devices from peer object.
+func FetchDevices(peerInfo *peer.Peer) ([]deviceapi.Info, error) {
```

Comment:
This function name is misleading. This could very well mean fetching it from store which is not true. How about naming this `GetDevicesFromPeer` or just `DevicesFromPeer`